src/aten_sink.erl

```diff
@@ -52,8 +54,13 @@ handle_call(get_failure_probabilities, From, State) ->
     % reply in a different process as we don't want calculation times
     % to affect the sample time of any incoming heartbeats
     _ = spawn(fun () ->
-                      Probs = get_probabilities(Data),
-                      gen_server:reply(From, Probs)
+                      try
+                          Probs = get_probabilities(Data),
+                          gen_server:reply(From, Probs)
+                      catch ErrType:Error ->
+                          ?LOG_ERROR("get_probabilities error: ~p:~P", [ErrType, Error, 10]),
+                          gen_server:reply(From, maps:new())
```

Comment:
could this affect the actual failure detection calculation? Maybe it is better not to reply at all and just log so that we have visibility of the actual error?

Reply:
Good idea, not replying will ensure the `timeout` happens, too and is logged.